roottest/root/treeformula/array/execOutOfBounds.C

```diff
@@ -7,8 +7,8 @@ int execOutOfBounds() {
   TTree *t; _file0->GetObject("t",t);
   TH2F *h2 = new TH2F("h2","h2",5,0,5,100,0,100);
   t->Draw("ThetaDeg[ringtouche_DE1]:ringtouche_DE1>>h2","","colz goff");
-  if ( (int)(h2->GetMean(1)*1000) != 3) {
-     printf("Error: x mean is %f rather than 0.003\n",h2->GetMean(1));
+  if ( (int)(h2->GetMean(1)*1000) != 0) {
+     printf("Error: x mean is %f rather than 0.000\n",h2->GetMean(1));
```

Comment:
Why is replacing `zeros` by `NaN` leads to a decrease in the value?

Reply:
```
root [0] auto h2 = new TH2F("h2", "h2", 100, 0, 1, 100, 0, 1);
root [1] h2->Fill(0.003, TMath::SignalingNaN())
(int) -1
root [2] h2->Fill(0.003, TMath::QuietNaN())
(int) -1
root [3] h2->GetMean(1)
(double) 0.0000000
```